### ISSUE TITLE
Does the FeCL loss in the code use the labels of unlabeled data?

### ISSUE THREAD

**Issue by regedit818** [KU-CVML/DyCON#9]
On line 226 of train_DyCON_Pancreas.py
`mask_con = F.avg_pool3d(label_batch.float(), kernel_size=args.feature_scaler*4, stride=args.feature_scaler*4)`
The 'label_batch' does not distinguish between labeled and unlabeled data.

In your other responses, you said that this is to construct positive and negative sample pairs, but doesn't this still involve the issue of data leakage?

**Comment by YcopymanY** [comment]
"Is the mask a pseudo-label generated by the teacher, or is it ground truth (GT)?"

**Comment by Endarzboy** [comment]
Hi @regedit818 

Sorry for the late reply.

Thank you, according to our preliminary experiments, the usage of masks for FeCL didn't cause any data leakage since it was optimized to enhance at the encoder-level, not at the task-level (decoder).

Best,

**Comment by Endarzboy** [comment]
Hi @DDDyylan 

Thanks for your interest in our work.

Please note that these are the labeled branch losess, so that the masks are necessary.
```
loss_seg = F.cross_entropy(stud_logits[:labeled_bs], label_batch[:labeled_bs]) # 0.9020 
loss_seg_dice = losses.dice_loss(stud_probs[:labeled_bs, 1, :, :, :], label_batch[:labeled_bs] == 1) # 0.9880
```

Best,

**Comment by Endarzboy** [comment]
Hi @YcopymanY  

Thanks

Kindly refer to issue #1, it's explained there.

Best,
